lib/posix-process/arch/x86_64/execve.c

```diff
@@ -0,0 +1,38 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/arch/ctx.h>
+#include <uk/essentials.h>
+
+void execve_arch_execenv_init(struct ukarch_execenv *execenv_new,
+			      struct ukarch_execenv *execenv,
+			      __uptr ip, __uptr sp)
+{
+	UK_ASSERT(execenv_new);
+	UK_ASSERT(execenv);
+	UK_ASSERT(ip);
+	UK_ASSERT(sp);
+	UK_ASSERT(IS_ALIGNED(sp, UKARCH_SP_ALIGN));
+
+	execenv_new->regs.rip = ip;
+	execenv_new->regs.rsp = sp;
+
+	/* Prepare for iretq
+	 * FIXME re-arch: use GDT macros once moved out of plat/common
+	 */
+	execenv_new->regs.eflags = X86_EFLAGS_IF; /* enable IRQs on ctx switch */
```

Comment:
I don't think this is necessary anymore: <https://github.com/unikraft/unikraft/blob/staging/plat/common/x86/syscall.S#L188>? Either way, not sure about direct assignment, but maybe a `|=` instead.

Reply:
The thing is that we need to initialize it to something as currently the new context contains whatever happens to be in the stack, where it's allocated from. And that's why we also can't do `|=`.

Reply:
What I will do is copy the state from `execenv` instead.